diff --git a/server/NatController.cpp b/server/NatController.cpp
index cda8f5f..670d50a 100644
--- a/server/NatController.cpp
+++ b/server/NatController.cpp
@@ -375,12 +375,12 @@ int NatController::setForwardRules(bool add, const char *intIface, const char *e
         rc = -1;
         goto err_return;
     }
-
+#ifndef TARGET_DISABLE_IPV6_RPFILTER
     if (runCmd(ARRAY_SIZE(cmd4), cmd4) && add) {
         rc = -1;
         goto err_rpfilter;
     }
-
+#endif
     if (setTetherCountingRules(add, intIface, extIface) && add) {
         rc = -1;
         goto err_return;
diff --git a/server/SoftapController.cpp b/server/SoftapController.cpp
index 27ed684..149d111 100755
--- a/server/SoftapController.cpp
+++ b/server/SoftapController.cpp
@@ -54,7 +54,11 @@ using android::base::WriteStringToFile;
 #endif
 
 #ifdef LIBWPA_CLIENT_EXISTS
+#ifdef MTK_HARDWARE
+static const char HOSTAPD_UNIX_FILE[]    = "/data/misc/wifi/hostapd/ap0";
+#else
 static const char HOSTAPD_UNIX_FILE[]    = "/data/misc/wifi/hostapd/wlan0";
+#endif
 static const char HOSTAPD_DHCP_DIR[]    = "/data/misc/dhcp";
 #endif
 static const char HOSTAPD_CONF_FILE[]    = "/data/misc/wifi/hostapd.conf";
@@ -282,7 +286,12 @@ int SoftapController::setSoftap(int argc, char *argv[]) {
             "hw_mode=%c\n"
             "ignore_broadcast_ssid=%d\n"
             "wowlan_triggers=any\n",
-            argv[2], argv[3], channel, (channel <= 14) ? 'g' : 'a', hidden));
+#ifdef MTK_HARDWARE
+	    "ap0",
+#else
+            argv[2],
+#endif
+	    argv[3], channel, (channel <= 14) ? 'g' : 'a', hidden));
 
     std::string fbuf;
     if (argc > 7) {
