From: daniel_hk (https://github.com/daniel_hk)
Date Tue, 17 Jan 2017 15:40:08 +0000
Subject: Skip the empty SSID in scan message

Add the length check to skip empty entries.

diff --git a/src/drivers/driver_nl80211_scan.c b/src/drivers/driver_nl80211_scan.c
index c115b6b..8347a7f 100644
--- a/src/drivers/driver_nl80211_scan.c
+++ b/src/drivers/driver_nl80211_scan.c
@@ -137,9 +137,10 @@ nl80211_scan_common(struct i802_bss *bss, u8 cmd,
                        wpa_hexdump_ascii(MSG_MSGDUMP, "nl80211: Scan SSID",
                                          params->ssids[i].ssid,
                                          params->ssids[i].ssid_len);
-                       if (nla_put(msg, i + 1, params->ssids[i].ssid_len,
-                                   params->ssids[i].ssid))
-                               goto fail;
+                       if (params->ssids[i].ssid_len)
+                               if (nla_put(msg, i + 1, params->ssids[i].ssid_len,
+                                           params->ssids[i].ssid))
+                                       goto fail;
                }
                nla_nest_end(msg, ssids);
        } else {

